### DISCUSSION TITLE
How to write a middleware that can read from HttpRequest and handle errors before responding to request client?

### DISCUSSION THREAD

**Discussion by 10fish** [actix/actix-web#2048] [Q&A] [answered]
As the title mentioned, I need middleware to handle common HTTP errors (404 / 500 / ) and it may result in url redirect, etc.

How to archive that? 

Or are there any other approaches to do this?

Thanks.

**Comment by robjtede** [comment, marked as answer]
Some resources to check out.

https://actix.rs/docs/middleware/

https://github.com/actix/examples/tree/master/basics/middleware

Also look at the implementations of the built in middleware: https://github.com/actix/actix-web/tree/master/src/middleware

**Reply by 10fish** [reply]
Thanks, I'll try with that.
